apps/frontend/src/components/searchBar.tsx

```diff
@@ -0,0 +1,310 @@
+          case "query":
+            searchObject.query = value;
+            break;
+          case "minPrice":
+            searchObject.minPrice = isNaN(+value) ? null : +value;
+            break;
+          case "maxPrice":
+            searchObject.maxPrice = isNaN(+value) ? null : +value;
+            break;
+          case "status":
+            searchObject.status = value;
+            break;
+          case "searchType":
+            searchObject.searchType = value;
+            break;
+          case "latitude":
+            searchObject.latitude = parseFloat(value);
+            break;
+          case "longitude":
+            searchObject.longitude = parseFloat(value);
+            break;
+          case "sort":
+            searchObject.sort = value;
+            break;
+          case "page":
+            searchObject.page = parseInt(value);
+            break;
+          case "limit":
+            searchObject.limit = parseInt(value);
+            break;
+          default:
+            break;
+        }
+      }
+    }
+    setSearchInput(searchObject.query);
+    setFilters(searchObject);
+  }, []);
+
+  const handleKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === "Enter") {
+      handleSearch();
+    }
+  };
+
+  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
+  const open = Boolean(anchorEl);
+  const handleClick = (event: React.MouseEvent<HTMLElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  const isDesktop = useMediaQuery("(min-width:1240px)");
+  const isSmallPage = useMediaQuery("(min-width:740px)");
+
+  return (
+    <Grid
+      container
+      direction="row"
+      alignItems="center"
+      justifyContent="center"
+      spacing={2}
+      sx={{
+        backgroundColor: "#f5f5f5",
+        boxShadow: "0px 4px 6px #808080",
+        paddingBottom: "10px",
+      }}
+    >
+      <Grid item>
+        {isSmallPage && (
+          <Button
+            variant="contained"
+            sx={classes.homepageButton}
+            onClick={handleReload}
+          >
+            <img
+              src={martletPlaceLogo}
+              alt="MartletPlace Logo"
+              style={{ width: "60px", marginBottom: "9px" }}
+            />
+            MartletPlace
+          </Button>
+        )}
+      </Grid>
+      <Grid item xs={6}>
+        <TextField
+          id="outlined-basic"
+          placeholder="Search"
+          variant="outlined"
+          fullWidth
+          sx={classes.searchBar}
+          value={searchInput}
+          onChange={handleSearchInputChange}
+          onKeyDown={handleKeyDown}
+        />
+      </Grid>
+      <Grid item>
+        <Button
+          type="submit"
+          variant="contained"
+          sx={{
+            ...classes.button,
+            ...classes.whiteButton,
+            backgroundColor: showFilters ? "#808080" : "white",
```

Comment:
Nit: Can we add #808080 as a color to the colors.tsx file? I think there is a `martletplaceWhite` in that file you can also use?

This is just a nice to have so others can use the same colors.